olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -673,8 +673,8 @@ ErrorCode NetworkCurl::SendImplementation(
   curl_easy_setopt(handle->handle, CURLOPT_FOLLOWLOCATION, 1L);
   curl_easy_setopt(handle->handle, CURLOPT_CONNECTTIMEOUT_MS,
                    config.GetConnectionTimeoutDuration().count());
-  curl_easy_setopt(handle->handle, CURLOPT_TIMEOUT_MS,
-                   config.GetTransferTimeoutDuration().count());
+  curl_easy_setopt(handle->handle, CURLOPT_SERVER_RESPONSE_TIMEOUT,
```

Comment:
Are you sure about this one?
It description looks more like connection timeout, and `CURLOPT_TIMEOUT_MS` is a transfer timeout.

Reply:
CURLOPT_TIMEOUT_MS sets overall time for request to finish, means it will not work as a transfer timeout. This one is actually what we need: "Causes libcurl to set a timeout period (in seconds) on the amount of time that the server is allowed to take in order to send a response message for a command before the session is considered dead." Since there is no other timeouts related to data transfer in curl.